packages/arb-avm-cpp/data_storage/src/arbcore.cpp

```diff
@@ -159,6 +157,81 @@ ValueLoader ArbCore::makeValueLoader() const {
         data_storage, core_code, ValueCache{1, 0})};
 }
 
+void ArbCore::printDatabaseMetadata() {
+    ReadTransaction tx(data_storage);
+    auto schema_version_result = schemaVersion(tx);
+    std::cout << "schema version:                   ";
+    if (!schema_version_result.status.ok()) {
+        std::cout << schema_version_result.status.ToString() << "\n";
+    } else {
+        std::cout << schema_version_result.data << "\n";
```

Comment:
Maybe a templated helper function to print out a generic DbResult could be helpful? The machine output would still be manual but everything else could be collapsed a lot, and inlined without assigning to a variable each time even.

Reply:
:+1: